### ISSUE TITLE
very long errors

### ISSUE THREAD

**Issue by bblay** [metomi/fab#164]
Make these error message more readable by printing the newlines as such and wrapping remaining long lines.

```
Traceback (most recent call last):
  File "/home/h01/jhickson/.conda/envs/sci-fab/lib/python3.10/site-packages/fab/build_config.py", line 96, in run
    step.run(artefact_store=artefact_store, config=self)
  File "/home/h01/jhickson/.conda/envs/sci-fab/lib/python3.10/site-packages/fab/steps/compile_fortran.py", line 63, in run
    check_for_errors(results_this_pass, caller_label=self.name)
  File "/home/h01/jhickson/.conda/envs/sci-fab/lib/python3.10/site-packages/fab/util.py", line 163, in check_for_errors
    raise RuntimeError(
RuntimeError: ('Error calling compiler:', RuntimeError("Command failed:\n['mpif90', '-c', '-I/project/jules/lib/ifort-16.0.1-gcc-4.8.5/netcdf-c-4.6.1-fortran-4.4.4-parallel/include', '-heap-arrays', '-std03', '-fpscomp', 'logicals', '-diag-disable', '6477', '-assume', 'nosource_include,protect_parens', '-fp-model', 'precise', '-no-vec', '-traceback', '/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90', '-o', '/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.o']\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(334): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [JULES_DEPOSITION_MOD]\nUSE jules_deposition_mod, ONLY: l_deposition, l_deposition_flux\n----^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(190): warning #7410: Fortran 2003 does not allow this keyword.   [DIRECTORY]\nINQUIRE(directory = output_dir, EXIST = dir_exists)\n--------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(696): error #6404: This name does not have a type, and must have an explicit type.   [L_DEPOSITION]\n  IF ( .NOT. l_deposition ) THEN\n-------------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(696): warning #6188: Fortran 2003 requires a LOGICAL data type in this context.   [L_DEPOSITION]\n  IF ( .NOT. l_deposition ) THEN\n-------------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(696): warning #6188: Fortran 2003 requires a LOGICAL data type in this context.\n  IF ( .NOT. l_deposition ) THEN\n-------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(709): error #6404: This name does not have a type, and must have an explicit type.   [L_DEPOSITION_FLUX]\n    IF ( .NOT. l_deposition_flux ) THEN\n---------------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(709): warning #6188: Fortran 2003 requires a LOGICAL data type in this context.   [L_DEPOSITION_FLUX]\n    IF ( .NOT. l_deposition_flux ) THEN\n---------------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(709): warning #6188: Fortran 2003 requires a LOGICAL data type in this context.\n    IF ( .NOT. l_deposition_flux ) THEN\n---------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(334): error #6580: Name in only-list does not exist.   [L_DEPOSITION]\nUSE jules_deposition_mod, ONLY: l_deposition, l_deposition_flux\n--------------------------------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(334): error #6580: Name in only-list does not exist.   [L_DEPOSITION_FLUX]\nUSE jules_deposition_mod, ONLY: l_deposition, l_deposition_flux\n----------------------------------------------^\ncompilation aborted for /home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90 (code 1)\n"))

1 error(s) found during compile fortran

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/spice/data/users/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/bin/build_jules.py", line 285, in <module>
    jules_config(revision=args.revision).run()
  File "/home/h01/jhickson/.conda/envs/sci-fab/lib/python3.10/site-packages/fab/build_config.py", line 100, in run
    raise Exception(f'\n\nError running build steps:\n{err}')
Exception: 

Error running build steps:
('Error calling compiler:', RuntimeError("Command failed:\n['mpif90', '-c', '-I/project/jules/lib/ifort-16.0.1-gcc-4.8.5/netcdf-c-4.6.1-fortran-4.4.4-parallel/include', '-heap-arrays', '-std03', '-fpscomp', 'logicals', '-diag-disable', '6477', '-assume', 'nosource_include,protect_parens', '-fp-model', 'precise', '-no-vec', '-traceback', '/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90', '-o', '/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.o']\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(334): error #7002: Error in opening the compiled module file.  Check INCLUDE paths.   [JULES_DEPOSITION_MOD]\nUSE jules_deposition_mod, ONLY: l_deposition, l_deposition_flux\n----^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(190): warning #7410: Fortran 2003 does not allow this keyword.   [DIRECTORY]\nINQUIRE(directory = output_dir, EXIST = dir_exists)\n--------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(696): error #6404: This name does not have a type, and must have an explicit type.   [L_DEPOSITION]\n  IF ( .NOT. l_deposition ) THEN\n-------------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(696): warning #6188: Fortran 2003 requires a LOGICAL data type in this context.   [L_DEPOSITION]\n  IF ( .NOT. l_deposition ) THEN\n-------------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(696): warning #6188: Fortran 2003 requires a LOGICAL data type in this context.\n  IF ( .NOT. l_deposition ) THEN\n-------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(709): error #6404: This name does not have a type, and must have an explicit type.   [L_DEPOSITION_FLUX]\n    IF ( .NOT. l_deposition_flux ) THEN\n---------------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(709): warning #6188: Fortran 2003 requires a LOGICAL data type in this context.   [L_DEPOSITION_FLUX]\n    IF ( .NOT. l_deposition_flux ) THEN\n---------------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(709): warning #6188: Fortran 2003 requires a LOGICAL data type in this context.\n    IF ( .NOT. l_deposition_flux ) THEN\n---------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(334): error #6580: Name in only-list does not exist.   [L_DEPOSITION]\nUSE jules_deposition_mod, ONLY: l_deposition, l_deposition_flux\n--------------------------------^\n/home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90(334): error #6580: Name in only-list does not exist.   [L_DEPOSITION_FLUX]\nUSE jules_deposition_mod, ONLY: l_deposition, l_deposition_flux\n----------------------------------------------^\ncompilation aborted for /home/h01/jhickson/cylc-run/trunk_24163/work/1/fab_jules_rivers_meto_linux_intel_normal_mpi_noomp/fab_workspace/jules/build_output/src/initialisation/standalone/init_output_mod.f90 (code 1)\n"))

1 error(s) found during compile fortran
```

Todo:
 - [ ] confirm fix with user

**Comment by bblay** [comment]
This is largely already fixed in v0.1.0. Testing with mocked output on the two versions:

### v0.9.0a1
```
Error running build steps:
('Error calling compiler:', RuntimeError('1\n    2\n    3\n    4\n    5\n'))
```

### v0.1.0
```
Error running build steps:
Error compiling /tmp/persistent/fab_workspace/gcom_object_archive_vn7.6_gfortran/build_output/gcom/gc/gc_kinds_mod.f90: 1
        2
        3
        4
        5    

```

I'll make a PR to put a newline before the `1`.

**Comment by bblay** [comment]
With regards to wrapping otherwise long lines, I suspect that's unhelpful as they can make the output unnavigable, as in the following example.

```
error running compiler: <really long command containing thousands of object files>
short error line 1
short error line 2
short error line 3

```

**Comment by bblay** [comment]
Resolved by #165
